### PR TITLE
Suggested Changes

### PR DESCRIPTION
**Pull request by goneneren**

### CODE REVIEW

File: .github/workflows/shared-build-deploy-container.yml [hunk 1]
```diff
@@ -114,7 +117,7 @@ jobs:
           echo "environment=$DEPLOYMENT_ENVIRONMENT" >> $GITHUB_OUTPUT
 
   validate-event:
-    name: "Block manual releases to staging and production"
+    name: "Block workflow_dispatch for staging and production"
```
Comment (emanuelgaspar) [inline .github/workflows/shared-build-deploy-container.yml:120]: `workflow_dispatch` in plain English means to run a workflow manually:
<https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow#configuring-a-workflow-to-run-manually>

A person reading this label must know that.
If they don't, having `workflow_dispatch` in the label is cryptic for no relevant reason.
I strongly prefer (and vote) to keep the "plain English" name here.

File: .github/workflows/shared-build-deploy-container.yml [hunk 2]
```diff
@@ -88,10 +89,12 @@ jobs:
           echo "Using runner: $RUNNER_NAME"
           echo "RUNNER_NAME=$RUNNER_NAME" >> $GITHUB_ENV
           echo "runnerName=$RUNNER_NAME" >> $GITHUB_OUTPUT
-
+  
+  # TODO: We use the same name for the workflow step and the step inside the workflow step which creates confusion
   pick-environment:
     name: "Pick environment"
     runs-on: ubuntu-latest
+    # TODO : pick-environment-step is not used anywhere
```
Comment (emanuelgaspar) [inline .github/workflows/shared-build-deploy-container.yml:97]: It's used in line 127:

```yml
needs.pick-environment.outputs.environment
```

Comment (emanuelgaspar) [reply .github/workflows/shared-build-deploy-container.yml:97]: The way to setup an `output` from a job requires the step to have an id:
<https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs>

It's verbose and convoluted, fully agree with you.
But I don't think we can avoid giving the inner step an `id`..

File: .github/workflows/shared-build-deploy-container.yml [hunk 3]
```diff
@@ -88,10 +89,12 @@ jobs:
           echo "Using runner: $RUNNER_NAME"
           echo "RUNNER_NAME=$RUNNER_NAME" >> $GITHUB_ENV
           echo "runnerName=$RUNNER_NAME" >> $GITHUB_OUTPUT
-
+  
+  # TODO: We use the same name for the workflow step and the step inside the workflow step which creates confusion
```
Comment (emanuelgaspar) [inline .github/workflows/shared-build-deploy-container.yml:93]: Here, we have:
- one job
- one step
- an id for a step

They are all for the same thing.
Should we add suffixes to all of them, to make it clearer?
```yml
pick-environment-job
pick-environment-step
pick-environment-step-id
```
?

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -1,3 +1,5 @@
+# TODO: name of this file is very confusing, we have in the project repos deploy.yml which calls the shared-build-deploy-container.yml
```
Comment (emanuelgaspar) [inline .github/workflows/deploy.yml:1]: This `deploy.yml` is meant to deploy this particular repo.
It's not a .NET solution, so it makes sense that it does different things.

Maybe we should refine the names in each repo:
`deploy-actions.yml`
`deploy-dotnet.yml`

Comment (joaofbvieira) [reply .github/workflows/deploy.yml:1]: Yes, exactly. Unfortunately we can't create nested folders here to organise better 😞

File: .github/workflows/shared-build-deploy-container.yml (hunk 3 above)
Comment (joaofbvieira) [reply .github/workflows/shared-build-deploy-container.yml:93]: This was refactored. Need to be re-reviewed 😅

File: .github/workflows/shared-build-deploy-container.yml (hunk 2 above)
Comment (joaofbvieira) [reply .github/workflows/shared-build-deploy-container.yml:97]: This was refactored. Need to be re-reviewed 😅 
Also we must use `needs`. We also renamed a bit this so maybe this isn't an issue anymore.

File: .github/workflows/shared-build-deploy-container.yml [hunk 5]
```diff
@@ -128,9 +131,25 @@ jobs:
             fi
           fi
 
+  build-test-cover-solution:
```
Comment (joaofbvieira) [inline .github/workflows/shared-build-deploy-container.yml:134]: I don't quite agree of having this job here, since we are already running this in our pull requests. So IMO it is just taking more time to deploy with something we already did. 🤔 

What are your thoughts about this?

**Review by joaofbvieira — COMMENTED** [review]
Left my 10 cents hehe 🤣 
I think after the merge with dev to review this PR again, since we made some improvements already.
